### ISSUE TITLE
Feature request: Support for $__SHREG_DFF_* in write_verilog

### ISSUE THREAD

**Issue by azonenberg** [cliffordwolf/yosys#374]
Muxes and other higher level cells are dumped as behavioral logic, not raw primitive instances. Can you add support for shift registers?

**Comment by cliffordwolf** [comment]
No, because `$__SHREG_DFF_*` are not Yosys standard cells (thus the double underscore). They are not supported by any of the internal frameworks (SAT solver, eval, sim lib, etc). They are only meant to be intermediate cells between `shregmap` and custom techmap rules.

This is similar to cells like `$__XILINX_RAM64X1D`, `$__DFFE_NN0`, `$__ICE40_RAM4K_M123`, and `$__inpad` that are used as intermediate cell types in various flows but are not Yosys standard cell types.

It would be possible to add shift register cells to the Yosys standard cell library. But that's either all or nothing. I.e. then there would need to be support code for the new cell type in all of Yosys.

For this I'd need at least a reasonable motivating example that demonstrates that a shift register cell type would be a necessary addition.
